code/modules/religion/rites.dm

```diff
@@ -102,3 +102,116 @@
+	favor_cost = 1000
+///the piece of clothing that will be fireproofed, only one per rite
+	var/obj/item/clothing/chosen_clothing
+
+/datum/religion_rites/fireproof/perform_rite(mob/living/user, atom/religious_tool)
+	for(var/obj/item/clothing/apparel in get_turf(religious_tool))
+		if(apparel.max_heat_protection_temperature >= FIRE_IMMUNITY_MAX_TEMP_PROTECT)
+			continue //we ignore anything that is already fireproof
+		chosen_clothing = apparel //the apparel has been chosen by our lord and savior
+		return ..()
+	return FALSE
+
+/datum/religion_rites/fireproof/invoke_effect(mob/living/user, atom/religious_tool)
+	for(chosen_clothing in get_turf(religious_tool))
+		if(istype(chosen_clothing,/obj/item/clothing/suit/hooded) || istype(chosen_clothing,/obj/item/clothing/suit/space/hardsuit ))
+			for(var/obj/item/clothing/head/integrated_helmet in chosen_clothing.contents) //check if the clothing has a hood/helmet integrated and fireproof it if there is one.
+				apply_fireproof(integrated_helmet)
+		apply_fireproof(chosen_clothing)
+		playsound(get_turf(religious_tool), 'sound/magic/fireball.ogg', 50, TRUE)
+		chosen_clothing = null //our lord and savior no longer cares about this apparel
+		return TRUE
+	chosen_clothing = null
+	to_chat(user, "<span class='warning'>The clothing that was chosen for the rite is no longer on the altar!</span>")
+	return FALSE
+
+
+/datum/religion_rites/burning_sacrifice
+	name = "Candle Fuel"
+	desc = "Sacrifice a buckled burning corpse for favor, the more burn damage the corpse has the more favor you will receive."
+	ritual_length = 20 SECONDS
+	ritual_invocations = list("To feed the fire of the one true flame ...",
+	"... to make it burn brighter ...",
+	"... so that it may consume all in its path ...",
+	"... I offer you this pitiful being ...")
+	invoke_msg = "... may it join you in the amalgamation of wax and fire, and become one in the black and white scene. "
+///the burning corpse chosen for the sacrifice of the rite
+	var/mob/living/carbon/chosen_sacrifice
+
+/datum/religion_rites/burning_sacrifice/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
+	if(!LAZYLEN(movable_reltool.buckled_mobs))
+		to_chat(user, "<span class='warning'>Nothing is buckled to the altar!</span>")
+		return FALSE
+	for(var/mob/living/carbon/corpse in movable_reltool.buckled_mobs) //works with any carbon corpse,not just humans
+		if(corpse.stat != DEAD)
+			to_chat(user, "<span class='warning'>You can only sacrifice dead bodies, this one is still alive!</span>")
+			return FALSE
+		if(!corpse.on_fire)
+			to_chat(user, "<span class='warning'>This corpse needs to be on fire to be sacrificed!</span>")
+			return FALSE
+		chosen_sacrifice = corpse
+		return ..()
```

Comment:
if you just made it any living mob (the type) you wouldnt have to type the loop. There also doesnt seem to be any feedback that the mob needs to be a carbon.

Reply:
I will have to disagree on this one, see if I don't check what is buckled with the for loop, then I can't confirm that it's a carbon, then I can't check for the other condition, so someone could buckle the corpse of a random mob, start the ritual, then swap that in with another corpse at the last moment. Now this is very unlikely to happen, but considering this is a free rite that dusts the sacrifice, I'd rather the sacrifice is done fair and square. And it's just a potential issue report waiting to happen anyway

I will however add a carbon check since it only make sense

Reply:
I think you misunderstood what I wanted or I wrote it shitty

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#istypeless-for-loops

the only thing that should be in this list is living mobs, so it doesnt make sense to type it unless you are REALLY deadset on keeping it carbon.

Reply:
okay done, I think I got confused by "didn't have to type the loop" so I just thought I had to remove the for loop entirely, should be fine now